src/lib/liquidity-curve.js

```diff
@@ -8,19 +8,38 @@ class LiquidityCurve {
   }
 
   setPoints (points) {
-    this.points = points
+    let prev
+    this.points = points.slice()
+    for (let i = 0; i < this.points.length; i++) {
+      let point = this.points[i]
+      if (point[0] < 0) throw new InvalidLiquidityCurveError('Curve has point with negative x-coordinate')
+      // If the y-coordinate is negative, pretend that it isn't for the sake of
+      // validating the increasing series. `Route#amountAt` never returns a negative value, anyway.
+      if (point[1] < 0) point = [point[0], 0]
+      if (prev && point[0] <= prev[0]) {
+        throw new InvalidLiquidityCurveError('Curve x-coordinates must strictly increase in series')
+      }
+      if (prev && point[1] < prev[1]) {
+        throw new InvalidLiquidityCurveError('Curve y-coordinates must increase in series')
+      }
+      prev = point
+    }
   }
 
   getPoints () {
     return this.points
   }
 
+  /**
+   * Note: LiquidityCurve#amountAt can return negative values, whereas Route#amountAt
+   * will only return >=0. This is because LiquidityCurve#amountAt is used in `join()`,
+   * while Route#amountAt is used externally.
+   */
   amountAt (x) {
```

Comment:
`Route#amountAt` is the one that is used externally, and it doesn't return negative values.

`LiquidityCurve#amountAt` needs to return negative values because `LiquidityCurve#join` uses it to find intersections between multiple curves, and those curves may have negative values.